segmentation/polus-pytorch-inference-plugin/src/data.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import pathlib
+import typing
+
+import bfio
+import numpy
+import torch
+
+import utils
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+    datefmt='%d-%b-%y %H:%M:%S',
+)
+logger = logging.getLogger("data")
+logger.setLevel(utils.POLUS_LOG)
+
+
+TILE_INDICES = typing.Tuple[int, int, int, int]  # (y_min, y_max, x_min, x_max)
+TILE_ITEM = typing.Tuple[
+    pathlib.Path,  # path to image
+    TILE_INDICES,  # tile indices in the image
+    torch.Tensor  # the tile itself
+]
+SCALABLE_BATCH = typing.Tuple[
+    typing.List[pathlib.Path],  # list of image paths
+    typing.List[TILE_INDICES],  # list of indices in the corresponding image.
+    torch.Tensor  # stacked batch of tiles shaped (B, 1, Y, X)
+]
+
+
+class ScalableDataset:
```

Comment:
Can we overload the pytorch Dataset class here? That way we can drop it into other torch tools.